code/modules/unit_tests/door_access.dm

```diff
@@ -0,0 +1,42 @@
+/// Tests to make sure door access works correctly.
+/datum/unit_test/door_access_check
+
+/datum/unit_test/door_access_check/Run()
+	var/mob/living/carbon/human/subject = allocate(/mob/living/carbon/human/consistent, run_loc_floor_bottom_left)
+	var/obj/machinery/door/airlock/instant/door = allocate(/obj/machinery/door/airlock/instant, run_loc_floor_bottom_left, EAST) //special subtype that just flips the density var on open() and close(), akin to a real airlock.
+
+	// First, test that someone without any access can open a door that doesn't have any access requirements. Let's test it via using the bumpopen() proc, called when someone bumps into the door.
+	door.bumpopen(subject)
+	TEST_ASSERT_EQUAL(door.density, FALSE, "Subject failed to open access-free airlock!")
+	door.close() // close it here as well
+
+	// Alright, now let's test that someone with access can open a door that requires access when only req_access is set.
+	subject.equipOutfit(/datum/outfit/job/assistant/consistent) // set up the outfit here to ensure the last check is pure.
+	var/obj/item/card/id/advanced/keycard = subject.wear_id
+
+	// Test two accesses at once just to make sure the script hasn't changed on us.
+	keycard.access = list(ACCESS_ENGINEERING, ACCESS_MAINT_TUNNELS)
+	door.req_access = list(ACCESS_ENGINEERING, ACCESS_MAINT_TUNNELS)
+	door.bumpopen(subject)
+	TEST_ASSERT_EQUAL(door.density, FALSE, "Subject with valid access failed to open airlock access-locked behind req_access!")
```

Comment:
BTW these should've been `TEST_ASSERT(!door.density, "text")`, makes a nicer message